### PR TITLE
Calculate the national percent share for COPD and Diabetes

### LINKED ISSUE #515: Investigate why BRFSS datasets aren't displaying properly
COPD and diabetes, the table isn't populating "share of diabetes cases"

let's investigate if 
1) we're actually pulling this data from the backend
2) why share of diabetes cases isn't working

### PR DESCRIPTION
**Pull request by joshzarrabi**
Resolves #515

### CODE REVIEW

File: frontend/src/data/utils/datasetutils.ts [hunk 1]
```diff
@@ -142,6 +142,17 @@ export function percent(numerator: number, denominator: number): number | null {
     : Math.round((1000 * numerator) / denominator) / 10;
 }
 
+/** Finds expected value of an ailment based on a population sample. */
+export function estimateTotal(
+  numerator: number,
```
Comment (vanshkumar) [inline frontend/src/data/utils/datasetutils.ts:147]: Might be clearer to rename the arguments to this function to sample_count, sample_population, and total_population

File: frontend/src/data/variables/BrfssProvider.ts [hunk 2]
```diff
@@ -29,7 +39,48 @@ class BrfssProvider extends VariableProvider {
...
+
+    let consumedDatasetIds = ["brfss"];
+
     if (breakdowns.geography === "national") {
+      // Because we add together the estimated state totals
+      // we need to get the acs state breakdown
+      acsBreakdowns.geography = "state";
+
+      const acsQueryResponse = await this.acsProvider.getData(
+        new MetricQuery(["population", "population_pct"], acsBreakdowns)
```
Comment (vanshkumar) [inline frontend/src/data/variables/BrfssProvider.ts:54]: Do you actually use population_pct from the ACS provider anywhere? If not, you can just not request this metric

File: frontend/src/data/variables/BrfssProvider.ts [hunk 3]
```diff
@@ -29,7 +39,48 @@ class BrfssProvider extends VariableProvider {
...
     df = this.renameGeoColumns(df, breakdowns);
 
+    // TODO How to handle territories?
+    let acsBreakdowns = breakdowns.copy();
+    acsBreakdowns.time = false;
+
+    let consumedDatasetIds = ["brfss"];
+
     if (breakdowns.geography === "national") {
+      // Because we add together the estimated state totals
```
Comment (vanshkumar) [inline frontend/src/data/variables/BrfssProvider.ts:49]: IMO a more descriptive comment on how the logic for this case works would be useful (e.g. BRFSS is a survey, sampling rates are diff across states, so we have to compute estimated state totals first)

**Review by vanshkumar — APPROVED** [review]
looks good!

File: frontend/src/data/utils/datasetutils.ts (hunk 1 above)
Comment (joshzarrabi) [reply frontend/src/data/utils/datasetutils.ts:147]: For sure

File: frontend/src/data/variables/BrfssProvider.ts (hunk 2 above)
Comment (joshzarrabi) [reply frontend/src/data/variables/BrfssProvider.ts:54]: Yeah I think I can get rid of it, thanks

File: frontend/src/data/variables/BrfssProvider.ts (hunk 3 above)
Comment (joshzarrabi) [reply frontend/src/data/variables/BrfssProvider.ts:49]: Hundo p

File: frontend/src/data/variables/BrfssProvider.ts [hunk 4]
```diff
@@ -29,7 +39,48 @@ class BrfssProvider extends VariableProvider {
...
+        estimated_total_copd: (row) =>
+          estimateTotal(
+            row.copd_count,
+            row.copd_count + row.copd_no,
+            row.population
+          ),
+      });
+
       df = df
         .pivot(breakdowns.demographicBreakdowns.race_and_ethnicity.columnName, {
```
Comment (vanshkumar) [inline frontend/src/data/variables/BrfssProvider.ts:85]: It's a bit odd how breakdowns.getSoleDemographicBreakdown().columnName is used in some places whereas breakdowns.demographicBreakdowns.race_and_ethnicity.columnName is used in others. Can you standardize this usage by defining a breakdownColumnName variable near the start of the function which is set to breakdowns.getSoleDemographicBreakdown().columnName, and using that throughout? That's what we do in other providers

**Review by vanshkumar — APPROVED** [review]

**Pull request merged**